server/integrationtests/local_permissions_test.go

```diff
@@ -0,0 +1,2047 @@
+		BlockPatches: []*model.BlockPatch{{Title: &newTitle}},
+	})
+
+	ttCases = []TestCase{
+		{"/boards-and-blocks", methodPatch, bab, userAnon, http.StatusUnauthorized, 0},
+		{"/boards-and-blocks", methodPatch, bab, userNoTeamMember, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodPatch, bab, userTeamMember, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodPatch, bab, userViewer, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodPatch, bab, userCommenter, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodPatch, bab, userEditor, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodPatch, bab, userAdmin, http.StatusOK, 1},
+	}
+	runTestCases(t, ttCases, testData, clients)
+}
+
+func TestLocalPermissionsDeleteBoardsAndBlocks(t *testing.T) {
+	th := SetupTestHelperLocalMode(t)
+	defer th.TearDown()
+	clients := setupLocalClients(th)
+	testData := setupData(t, th)
+
+	bab := toJSON(t, model.DeleteBoardsAndBlocks{
+		Boards: []string{testData.publicBoard.ID},
+		Blocks: []string{"block-3"},
+	})
+
+	ttCases := []TestCase{
+		{"/boards-and-blocks", methodDelete, bab, userAnon, http.StatusUnauthorized, 0},
+		{"/boards-and-blocks", methodDelete, bab, userNoTeamMember, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodDelete, bab, userTeamMember, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodDelete, bab, userViewer, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodDelete, bab, userCommenter, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodDelete, bab, userEditor, http.StatusForbidden, 0},
+		{"/boards-and-blocks", methodDelete, bab, userAdmin, http.StatusOK, 0},
+	}
+	runTestCases(t, ttCases, testData, clients)
+}
+
+func TestLocalPermissionsLogin(t *testing.T) {
+	th := SetupTestHelperLocalMode(t)
+	defer th.TearDown()
+	clients := setupLocalClients(th)
+	testData := setupData(t, th)
+
+	loginReq := func(username, password string) string {
+		return toJSON(t, api.LoginRequest{
+			Type:     "normal",
+			Username: username,
+			Password: password,
+		})
+	}
+
+	ttCases := []TestCase{
+		{"/login", methodPost, loginReq(userAnon, password), userAnon, http.StatusUnauthorized, 0},
+		{"/login", methodPost, loginReq(userAdmin, password), userAdmin, http.StatusOK, 1},
+	}
+	runTestCases(t, ttCases, testData, clients)
+}
+
+func TestLocalPermissionsLogout(t *testing.T) {
+	th := SetupTestHelperLocalMode(t)
+	defer th.TearDown()
+	clients := setupLocalClients(th)
+	testData := setupData(t, th)
+
+	ttCases := []TestCase{
+		{"/logout", methodPost, "", userAnon, http.StatusUnauthorized, 0},
+		{"/logout", methodPost, "", userAdmin, http.StatusOK, 0},
+	}
+	runTestCases(t, ttCases, testData, clients)
+}
+
+func TestLocalPermissionsRegister(t *testing.T) {
+	th := SetupTestHelperLocalMode(t)
+	defer th.TearDown()
+	clients := setupLocalClients(th)
+	testData := setupData(t, th)
+
+	ttCases := []TestCase{
+		{"/register", methodPost, "", userAnon, http.StatusInternalServerError, 0},
```

Comment:
Why register method is generating an `InternalServerError`?

Reply:
Thanks. I've updated the test case to make a successful register call.